src/Http/Controllers/Corporation/ContractsController.php

```diff
@@ -84,6 +86,54 @@ public function getContractsData(int $corporation_id)
 
                 return view('web::partials.contractcontentsbutton', compact('row'));
             })
+            ->filterColumn('issuer_id', function ($query, $keyword) {
+
+                $resolved_ids = UniverseName::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($resolved_id) {
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.issuer_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
+            })
+            ->filterColumn('assignee_id', function ($query, $keyword) {
+
+                $resolved_ids = UniverseName::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($resolved_id) {
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.assignee_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
+            })
+            ->filterColumn('acceptor_id', function ($query, $keyword) {
+
+                $resolved_ids = UniverseName::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($resolved_id) {
+                        return $resolved_id->entity_id;
+                    });
+
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')
+                    ->get()
+                    ->map(function ($character_info) {
+                        return $character_info->character_id;
+                    });
+
+                $query->whereIn('a.acceptor_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray()));
```

Comment:
can't we use collection merge directly instead splitting the two upper and making a final array merge based on array ?